File: shopee/2a.py
import pandas as pd
from sqlite3 import connect
import matplotlib.pyplot as plt

path_od = "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/order_detail.csv"
path_pd = "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/payment_detail.csv"
path_cd = "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/customer_detail.csv"
path_sd = "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/sku_detail.csv"
df_od = pd.read_csv(path_od)
df_pd = pd.read_csv(path_pd)
df_cd = pd.read_csv(path_cd)
df_sd = pd.read_csv(path_sd)

df = pd.merge(df_od, df_sd, left_on='sku_id', right_on='id', how="inner")
df = df[df['is_valid'] == 1]
df['order_date'] = pd.to_datetime(df['order_date'])
df['year'] = df['order_date'].dt.year

tabel = df.groupby(['category', 'year'])['qty_ordered'].sum().unstack()


def perubahan(x,y):
    return y-x
tabel['Perubahan'] = tabel.apply(lambda row: perubahan(row[2021], row[2022]), axis=1)

def keterangan(x):
    if x > 0:
        return 'Naik'
    elif x < 0:
        return 'Turun'
tabel['Keterangan'] = tabel['Perubahan'].apply(keterangan)
tabel = tabel.sort_values('Perubahan', ascending=False)
print(tabel)
print()


print('TOP 20 Produk Mengalami Penurunan')
df = df[df['category'].str.lower() == 'others']
tabel_akhir = df.groupby(['sku_name', 'year'])['qty_ordered'].sum().unstack().fillna(0)
tabel_akhir['Perubahan'] = tabel_akhir.apply(lambda row: perubahan(row[2021], row[2022]), axis=1)
tabel_akhir['Keterangan'] = tabel_akhir['Perubahan'].apply(keterangan)
tabel_akhir = tabel_akhir.sort_values('Perubahan', ascending=True)
print(tabel_akhir.head(20))

tabel_akhir.to_excel('Top 20.xlsx', index=True)

File: Analisis_data_shopee/2a.r
library(RMySQL)
library(dplyr)
library(tidyr)
library(lubridate)

path_od <- "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/order_detail.csv"
path_pd <- "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/payment_detail.csv"
path_cd <- "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/customer_detail.csv"
path_sd <- "https://raw.githubusercontent.com/dataskillsboost/FinalProjectDA11/main/sku_detail.csv"
df_od <- read.csv(path_od)
df_pd <- read.csv(path_pd)
df_cd <- read.csv(path_cd)
df_sd <- read.csv(path_sd)

df <- inner_join(df_od, df_sd, by = c("sku_id" = "id"))
df$order_date <- as.Date(df$order_date)
df <- filter(df, is_valid == 1)

tabel21 <- df[year(df$order_date) == 2021, ]
tabel21 <- tabel21 %>%
    group_by(category) %>%
    summarise(jumlah21 = sum(qty_ordered))
tabel22 <- df[year(df$order_date) == 2022, ]
tabel22 <- tabel22 %>%
    group_by(category) %>%
    summarise(jumlah22 = sum(qty_ordered))
tabel_akhir <- merge(tabel21, tabel22, by = "category")
tabel_akhir$perbedaan <- tabel_akhir$jumlah22 - tabel_akhir$jumlah21
tabel_akhir <- tabel_akhir %>% mutate(
    penjelasan = case_when(
        perbedaan > 0 ~ "Naik",
        perbedaan < 0 ~ "Turun",
        .default = NULL
    )
) %>% arrange(perbedaan)
print(tabel_akhir)
cat('\n\n\n\n\n\n\n')



print('tabel khusus kategori others')
df <- filter(df, category == 'Others')
tabel21 <- df[year(df$order_date) == 2021, ]
tabel21 <- tabel21 %>%
    group_by(sku_name) %>%
    summarise(jumlah21 = sum(qty_ordered))
tabel22 <- df[year(df$order_date) == 2022, ]
tabel22 <- tabel22 %>%
    group_by(sku_name) %>%
    summarise(jumlah22 = sum(qty_ordered))
tabel_akhir <- merge(tabel21, tabel22, by = "sku_name", all = TRUE)
tabel_akhir[is.na(tabel_akhir)] <- 0
tabel_akhir$perbedaan <- tabel_akhir$jumlah22 - tabel_akhir$jumlah21
tabel_akhir <- tabel_akhir %>% mutate(
    penjelasan = case_when(
        perbedaan > 0 ~ "Naik",
        perbedaan < 0 ~ "Turun",
        .default = NULL
    )
) %>% arrange(perbedaan)
print(head(tabel_akhir, 20))